### PR TITLE
display-config

### PR DESCRIPTION
**Pull request by AlanGriffiths**
Add logic for an optional "display-config" setting.

If "snapctl get display-config" is set we compare with the `.display` file (if it exists) and unless they are the same overwrite and restart.

### CODE REVIEW

**Comment by AlanGriffiths** [comment]
Oops. Sorry broke the logic.

File: snap/hooks/configure [hunk 1]
```diff
@@ -12,6 +12,27 @@ EOT
 let changes+=1
 fi
 
+# display-config
+display_config=$(snapctl get display-config)
+if [ -n "${display_config}" ]; then
```
Comment (Saviq) [inline snap/hooks/configure]: Hmm so if we set it to empty, the file will remain, along with the "DO NOT EDIT" header... Maybe we could check if the file was written by us (i.e. has the `# DO NOT EDIT` header) and remove it if that's the case?

File: snap/hooks/configure [hunk 2]
```diff
@@ -12,6 +12,27 @@ EOT
...
+  echo "# DO NOT EDIT THIS FILE BY HAND -- YOUR CHANGES WILL BE OVERWRITTEN" > "$SNAP_DATA/miral-kiosk.display~"
+  echo "# USE 'snap set mir-kiosk display-config=...' INSTEAD"              >> "$SNAP_DATA/miral-kiosk.display~"
+  echo ""                                                                   >> "$SNAP_DATA/miral-kiosk.display~"
+  echo "${display_config}"                                                  >> "$SNAP_DATA/miral-kiosk.display~"
```
Comment (Saviq) [inline snap/hooks/configure]: I suppose `$SNAP_DATA/miral-kiosk.display` deserves a variable of its own?

**Review by Saviq — CHANGES_REQUESTED** [review]
Just a couple nits

**Review by Saviq — APPROVED** [review]
Yeah this is fine, although screams for refactoring.

**Pull request merged**